genrl/agents/deep/ddpg/ddpg.py

```diff
@@ -62,7 +62,23 @@ def _create_model(self) -> None:
                 np.zeros_like(action_dim), self.noise_std * np.ones_like(action_dim)
             )
 
-        if isinstance(self.network, str):
+        if isinstance(self.network, str) and self.shared_layers is not None:
+            self.ac = get_model("ac", self.network + "s")(
+                state_dim,
+                action_dim,
+                self.shared_layers,
+                self.policy_layers,
+                self.value_layers,
+                "Qsa",
+                False,
+            ).to(self.device)
+            actor_params = list(self.ac.actor.parameters()) + list(
+                self.ac.shared.parameters()
+            )
+            critic_params = list(self.ac.critic.parameters()) + list(
+                self.ac.shared.parameters()
+            )
+        elif isinstance(self.network, str) and self.shared_layers is None:
```

Comment:
Try not to use these extra statements. Incorporate it with the normal actor critic initialization. It doesn't look good having this much extra code just for changing the architecture a little. Applies for all the agents

Reply:
It's adding about 20 lines total for every algorithm file, that should be avoided unless the change is major